### ISSUE TITLE
State of this project

### ISSUE THREAD

**Issue by kummerer94** [Logitech/logiPy#8]
Hi,

this looks like a great project! However, there has not been an update for a few years now so I wonder whether this code still actually works.

I wanted to control the LEDs of my logitech mouse and keyboard in Python. This was the only project I came across. Somehow, I could not make it work. Is there still someone maintaining this?

What would also be great is a few instructions on how to install the dependencies this project needs. I struggled to get Logitech Gaming Software to work on my computer (it would not detect any peripherals).

Thank you
Alexander

**Comment by remictrr** [comment]
I also wanted to control my keyboard lightning with python adn came across this project. It still work for me. for it to work you need to install the LogitechLedEnginesWrapper.dll from the link in the readme file and set its path in the logi_led.py script in the part where the dll is opened.

**Comment by kummerer94** [comment]
Thank you for the kind reply!

I just tried downloading the [LED Illumination SDK](https://www.logitechg.com/en-us/innovation/developer-lab.html). I found the LogitechLedEnginesWrapper.dll and specified the path in the `logi_led.py` script. Still, it still is not working.

I have Logitech G Hub installed and nothing else. Could this be the issue? Which kind of Logitech software do you have installed?

**Comment by remictrr** [comment]
yes i have GHUB installed and it's not an issue. maybe your issue is that logi_led.py can't find the dll. have you specified a relative path or an absolute path for the dll ? You need to use an absolute path. 
i used this line of code to get the dll path which is in the same folder as the logi_led.py script:
dll_path = os.path.join(os.path.dirname(__file__), 'LogitechLedEnginesWrapper.dll')

**Comment by dieser-niko** [comment]
There's a difference between using G Hub, Logitech Gaming Software and the LED Illumination SDK. After I tried to replicate the issue of @kummerer94 I've installed both softwares to check, which .dll files are being used. I guess, that the LED Illumination SDK and Logitech Gaming Software have some missing elements, at least that's what I'm thinking by looking at the sizes of those files. 

Here for comparison:
(all tested with x64)

Source | Size in KB | Filename | Can be loaded with ctypes
--- | --- | --- | ---
LED Illumination SDK | 20 | LogitechLedEnginesWrapper.dll | no
Logitech Gaming Software | 161 | LogitechLed.dll | no
G HUB | 4070 | sdk_legacy_led_x64.dll | yes

It actually suprises me that the Logitech Gaming Software (which LogiPy is even referring to) doesn't work. My guess is, that there are certain things that changed overtime and weren't implemented in this project, but I'm no expert, especially about .dll files. Also I don't have the time for testing out older softwares.

If you've already got G HUB and want to use that SDK, just look into https://github.com/Logitech/logiPy/pull/9, at least that worked for me

**Comment by kummerer94** [comment]
@dieser-niko I did not see this PR until now. Great idea. I will try it out!

**Comment by kummerer94** [comment]
@dieser-niko Thank you again for the hint. The library now works great for me. I will try to rewrite the code a bit to provide a better experience and API. (But I had to remove the ARX code since the app on iOS has not received any updates for 5 years and I have never used it.)

You can find my efforts here: https://github.com/kummerer94/logiPy
